playwright/components/link-preview/index.ts

```diff
@@ -0,0 +1,30 @@
+import type { Page } from "@playwright/test";
+import { PILL_CLOSE_ICON } from "../locators";
+import { dlsRoot } from "../../../playwright/components/index";
+import { LinkPreviewProps } from "../../../src/components/link-preview";
+
+// component preview locators
+export const linkPreview = (page: Page) => {
+  return dlsRoot(page).locator("a");
+};
+
+export const linkPreviewAs = (page: Page, as: LinkPreviewProps["as"]) => {
+  let selector: string;
+
+  switch (as) {
+    case "a":
+      selector = "a";
+      break;
+    case "div":
+      selector = "div";
+      break;
+    default:
+      throw new Error(`Invalid 'as' value: ${as}`);
+  }
+
+  return dlsRoot(page).locator(selector).first();
```

Comment:
As above, I don't think `dlsRoot(page)` needs to be specified here. You can just have `return locator(selector).first();` I think.